### ISSUE TITLE
Change `as_<type>()` to `get<type>()`

### ISSUE THREAD

**Issue by jkeiser** [simdjson/simdjson#607]
All the other JSON frameworks use `get()`.

```c++
auto [value, error] = element.
auto [value, error] = element.get<uint64_t>(); // New
```

It's also probably a nice idea to have a `get<uint64_t>("1/tire_pressure/0")` variant that follows a JSON pointer and casts the result; I find myself doing those two operations in tandem a lot. Pretty much any method that has an element could be modified with a templatized cast parameter, and it would help ergonomically for error code users.

Maybe also `is<uint64_t>()`, but I'm not as sure there.

I can imagine keeping `get_array()` and `get_object()` for convenience, though, since their names are `document::array` and `document::object`.

**Comment by lemire** [comment]
Sure... this seems minor... My only concern is that this should be settled before 0.3.

**Comment by jkeiser** [comment]
Fixed by #643.
